src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -296,9 +305,16 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                             if (change.ChangeType == KeyValueChangeType.Deleted)
                             {
                                 applicationSettings.Remove(change.Key);
+                                mappedData.Remove(change.Key);
                             }
                             else if (change.ChangeType == KeyValueChangeType.Modified)
                             {
+                                ConfigurationSetting setting = change.Current;
+                                foreach (Func<ConfigurationSetting, ValueTask<ConfigurationSetting>> func in _options.UserDefinedMappers)
+                                {
+                                    setting = func(setting).Result;

```

Comment:
The mapper will change the original ConfigurationSetting object, so mappedData and applicationSettings will both point to the modified object. This would be an issue when we want to inspect the value of the original  ConfigurationSetting object during refresh.